apollo-client/src/main/java/com/ctrip/framework/apollo/spring/boot/ApolloApplicationContextInitializer.java

```diff
@@ -123,7 +131,12 @@ protected void initialize(ConfigurableEnvironment environment) {
     logger.debug("Apollo bootstrap namespaces: {}", namespaces);
     List<String> namespaceList = NAMESPACE_SPLITTER.splitToList(namespaces);
 
-    CompositePropertySource composite = new CompositePropertySource(PropertySourcesConstants.APOLLO_BOOTSTRAP_PROPERTY_SOURCE_NAME);
+    CompositePropertySource composite;
+    if (configUtil.isPropertyNamesCacheEnabled()) {
```

Comment:
Please add unit tests to verify these 2 different scenarios, e.g. to check whether the property source is of `CompositeConfigPropertySource` type or `CompositePropertySource`.